code/wdb/datastripe.q

```diff
@@ -33,7 +33,7 @@ upserttopartition:{[dir;tablename;keycol;enumdata;nextp]
 	/- get unique sym from table
 	s:first raze value'[?[enumdata;();1b;enlist[keycol]!enlist keycol]];
 	/- get process specific taildir location
-	dir:` sv dir,.proc.procname,`;
+	dir:` sv .ds.td,(`$raze string `TDB,.ds.segmentid),`;
```

Comment:
```suggestion
	dir:` sv .ds.td,(`$"TDB",string .ds.segmentid),`;
```

think this will be the same but less messy

is there a reason to switch away from using procname? also this no longer uses the `dir` that gets passed into function, is that intentional? should that arg be removed?

Reply:
Sorry, it should be using dir instead of .ds.td. 
Your line to create the directory however creates a list of syms like so:
``q)`$"TDB",string .ds.segmentid``
 `` `T`D`B`1``

Reply:
It is a long

Reply:
It is a list of longs
``q)type .ds.segmentid``
``7h``

Reply:
why is it a list? surely there's only one id for a given process? seems strange